ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -244,7 +245,18 @@ - (void)prepareView {
 }
 
 - (BOOL)shouldContinue {
-    return ![self correctValueIfNeeded];
+    ORKTextAnswerFormat *answerFormat = (ORKTextAnswerFormat *)[self.step impliedAnswerFormat];
+    if (answerFormat.isEmail) {
+        BOOL isValid = [answerFormat isAnswerValidWithString:self.answer];
+
+        if (!isValid){
+            [self showValidityAlertWithMessage:self.answer];
+        }
+        
+        return isValid;
+    } else {
+        return ![self correctValueIfNeeded];
```

Comment:
I think we should cut the text from the end as it is right now, but also add an alert message notifying the reason for the cut. What do you think? Should I proceed with that implementation?